### PR TITLE
docs(releasing): Backport changes to docs for transform removal

### PR DESCRIPTION
**Pull request by jszwedko**
Extracted from f262fea3c

Noticed via https://github.com/vectordotdev/vector/discussions/13520#discussioncomment-3138619

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: website/content/en/docs/reference/configuration/_index.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ sources:
       - /var/log/apache2/*.log
     ignore_older: 86400
 transforms:
-  apache_parser:
+  remap:
```
Comment (spencergilbert) [inline website/content/en/docs/reference/configuration/_index.md:81]:
```suggestion
  apache_parser:
```
This is the ID, not the type!

File: website/content/en/docs/reference/configuration/_index.md [hunk 2]
```diff
@@ -125,7 +125,7 @@ sinks:
     }
   },
   "transforms": {
-    "apache_parser": {
+    "remap": {
```
Comment (spencergilbert) [inline website/content/en/docs/reference/configuration/_index.md:128]:
```suggestion
    "apache_parser": {
```
ID not type here as well 😃

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: website/content/en/docs/reference/configuration/_index.md (hunk 1 above)
Comment (jszwedko) [reply website/content/en/docs/reference/configuration/_index.md:81]: Ah, true, this needs to be fixed on `master` too. These changes were directly from https://github.com/vectordotdev/vector/commit/f262fea3cbf09ab8bfa1bcc4937964310d4b54e0

Comment (jszwedko) [reply website/content/en/docs/reference/configuration/_index.md:81]: Ah, actually a contributor fixed it in 96973fa24. I'll just cherry-pick that commit over too once this is merged.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**